tests/infra.py

```diff
@@ -22,6 +22,21 @@ def get_env(varname):
     return os.environ[varname]
 
 
+class ExpectedErrorFields(typing.NamedTuple):
+    code: str
+
+    status: typing.Optional[int] = None
+    """
+    If this is None, then we not check the status.  For all other values, we test that it matches.
+    """
+
+    expect_stacktrace: typing.Optional[bool] = None
+    """
+    If this is True, then we expect the stacktrace to be present.  If this is False, then we expect the stacktrace to be
+    absent.  If this is None, then we do not test the presence of the stacktrace.
+    """
```

Comment:
Just tested it with Sphinx and indeed, attribute docstrings go under the attribute declarations. Weird.